src/lib/mdns/DiscoveryManager.cpp

```diff
@@ -191,9 +198,21 @@ CHIP_ERROR DiscoveryManager::PublishUnprovisionedDevice(chip::Inet::IPAddressTyp
     entries[1].mDataSize = strnlen(discriminatorBuf, sizeof(vendorProductBuf));
 
 #if CHIP_ENABLE_ADDITIONAL_DATA_ADVERTISING
+    // retrieve serial number
+    SuccessOrExit(error =
+                      chip::DeviceLayer::ConfigurationMgr().GetSerialNumber(serialNumber, sizeof(serialNumber), serialNumberSize));
+    serialNumber[serialNumberSize] = '\0';
```

Comment:
I'm pretty sure GetSerialNumber() does this for you

Reply:
good point, fixed